AzureFirewall/AzureFirewallApplicationRule.kql

```diff
@@ -0,0 +1,38 @@
+let applicationRuleAllowsHttp = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Url: " url ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow"
+| project-away msg_s;
+let applicationRuleAllowsHttps = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow" and port_destination !hasprefix "80" and rule !has "Web Category"

```

Comment:
The union branch qualifying filters are not mutually exclusive. This will lead to duplication of events, even if the parse statement itself fails for one of the branches.

Reply:
I've now used `parse-where` and made sure I didn't miss anything by applying this on larger datasets (2 week of Azure Firewall logs) and comparing it with unparsed data.